### PR TITLE
incorrect_parameter_updates_#12

### PR DESCRIPTION
**Pull request by UmmulkiramR**
relay will now check for the availability of parameters and engine params to avoid null updates to the workflow ES document.

### CODE REVIEW

File: src/main/java/org/icgc_argo/workflow/relay/util/NextflowDocumentConverter.java [hunk 1]
```diff
@@ -56,19 +58,30 @@ public static WorkflowDocument buildWorkflowDocument(
...
+                : oldDocument.getEngineParameters().getResume())
+            .launchDir(ObjectUtils.isNotEmpty(workflow.getLaunchDir())
+                ? workflow.getLaunchDir()
+                : oldDocument.getEngineParameters().getLaunchDir())
+            .projectDir(ObjectUtils.isNotEmpty(workflow.getProjectDir())
+                ? workflow.getProjectDir()
+                : oldDocument.getEngineParameters().getProjectDir())
+            .workDir(ObjectUtils.isNotEmpty(workflow.getWorkDir())
+                ? workflow.getWorkDir()
+                : oldDocument.getEngineParameters().getWorkDir())
```
Comment (joneubank) [inline src/main/java/org/icgc_argo/workflow/relay/util/NextflowDocumentConverter.java:75]: This uses a lot of repeated code that makes it hard to read the actual logic. Can we use a method like `defaultIfNull` instead? 
example:
```java
.revision(ObjectUtils.defaultIfNull(workflow.getRevision(), oldDocument.getEngineParameters().getRevision())
```

If we need to check for empty as well as null, can we make our own util method `defaultIfEmpty`

**Review by joneubank — CHANGES_REQUESTED** [review]
Suggestion for cleaning up repeated code.

File: src/main/java/org/icgc_argo/workflow/relay/util/NextflowDocumentConverter.java (hunk 1 above)
Comment (joneubank) [reply src/main/java/org/icgc_argo/workflow/relay/util/NextflowDocumentConverter.java:75]: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/ObjectUtils.html#defaultIfNull-T-T-

Comment (UmmulkiramR) [reply src/main/java/org/icgc_argo/workflow/relay/util/NextflowDocumentConverter.java]: That makes so much sense. Made it concise. Hope this one is more readable.

File: src/main/java/org/icgc_argo/workflow/relay/util/ObjectUtilities.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.icgc_argo.workflow.relay.util;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+public class ObjectUtilities {
+
+  public static <T> T defaultIfNullOrEmpty(final T object, final T defaultValue){
+    return ObjectUtils.isNotEmpty(object)?object:defaultValue;
+  }
```
Comment (joneubank) [inline src/main/java/org/icgc_argo/workflow/relay/util/ObjectUtilities.java:9]: :+1: :pray:

**Review by joneubank — APPROVED** [review]

**Pull request merged**